src/Parser.js

```diff
@@ -402,7 +402,9 @@ export default class Parser {
             }
         }
 
-        if (superscript || subscript) {
+        // Base must be set if superscript or subscript are set per logic above,
+        // but need to check here for type check to pass.
+        if ((superscript || subscript) && base) {
```

Comment:
+kevinb
This is the failure
```
\begin{matrix}
    ^3+[-1][1-1]1=1(=1)\lvert a\rvert~b \\
    \scriptstyle{^3+[-1][1-1]1=1(=1)\lvert a\rvert~b} \\
    \scriptscriptstyle{^3+[-1][1-1]1=1(=1)\lvert a\rvert~b} \\
\end{matrix}
```
and I can see that would fail with this change in the code. This examples has a superscript `^3` without a base! Isn't that supposed to be a parse failure in LaTeX? Why do we support it?

Reply:
Interestingly, there are already tests for this and this is apparently a feature. I'll check to see why they didn't fail.

Reply:
Fixed and pushed.

Reply:
Sorry for the delay response.  I tested the above code on quicklatex.com and it works.  Thanks for fixing the issue.